src/PrestaShopBundle/Form/Admin/Extension/FormThemeExtension.php

```diff
@@ -0,0 +1,136 @@
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Extension;
+
+use Symfony\Component\Form\AbstractTypeExtension;
+use Symfony\Component\Form\Extension\Core\Type\FormType;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormRendererInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * This form extension introduce a form_theme option that can be used on any form (it's more usually used
+ * on compound forms), this allows setting the form theme in the form type options instead of the twig
+ * template. Thus adding this option on a form:
+ *
+ *     [
+ *         'form_theme' => '@PrestaShop/Admin/Sell/Catalog/Product/FormTheme/features.html.twig',
+ *     ]
+ *
+ * is equivalent to doing this in the twig template:
+ *
+ *     {% form_theme form '@PrestaShop/Admin/Sell/Catalog/Product/FormTheme/features.html.twig' %}
+ *
+ * This option is very handy to customize collection sub-types (or any other container form type), or if
+ * a module wishes to add its custom form type that needs a custom design. Since the rendering of the recent
+ * form is automatic with a single form_row it is hard to define the form theme from twig, that's one of the
+ * use cases of this extension.
+ *
+ * Additionally, you can also specify multiple form themes for example:
+ *
+ *     [
+ *         'form_theme' => [
+ *             'bootstrap_4_layout.html.twig',
+ *             '@PrestaShop/Admin/Sell/Catalog/Product/FormTheme/features.html.twig',
+ *         ],
+ *     ]
+ *
+ * is equivalent to:
+ *
+ *     {% form_theme productForm with ['bootstrap_4_layout.html.twig', '@PrestaShop/Admin/Sell/Catalog/Product/FormTheme/features.html.twig'] %}
+ *
+ *
+ * Finally, you should know that by default the form theme applied in tig are not complete replacement, they are actually
+ * applied on top of the default themes set in the app/config/config.yml file in the "twig > form_themes" section. In some cases
+ * you don't want to extend the theme from config you want to use your theme and your theme ONLY. In this case you can use the
+ * "use_default_themes" option to disable this behavior (by default it is true so config theme is always used by default).
+ *
+ * So, as an example this configuration:
+ *
+ *     [
+ *         'form_theme' => '@PrestaShop/Admin/Sell/Catalog/Product/FormTheme/product.html.twig',
+ *         'use_default_themes' => false,
+ *     ]
+ *
+ * is equivalent to the twig (the important thing here is the use of the "only" keyword):
+ *
+ *     {% form_theme productForm with ['@PrestaShop/Admin/Sell/Catalog/Product/FormTheme/product.html.twig'] only %}
```

Comment:
When this PR will be merged, shouldn't you add all this info to devdoc v8 ?

Maybe adding a section in this documentation ? 
https://devdocs.prestashop.com/8/modules/concepts/forms/admin-forms/

Reply:
Yep, it's planned as for all the other extensions we added during the product page development
I also planned to document the whole form structure once it's finalized, I already started this word for 178 but it will need many updates for 8.0 (I'll add the docu label on this PR just in case)